src/Algebra/Graph/AdjacencyMap.hs

```diff
@@ -117,9 +117,7 @@ overlays = AM . Map.unionsWith Set.union . map adjacencyMap
 -- 'isEmpty' . connects == 'all' 'isEmpty'
 -- @
 connects :: Ord a => [AdjacencyMap a] -> AdjacencyMap a
-connects []     = empty
-connects [x]    = x
-connects (x:xs) = x `connect` connects xs
+connects = foldr connect empty
```

Comment:
Here is fine, since there is nothing more clever.

Reply:
Although in future we might add a more efficient `connects` for adjacency maps too.